WordPress/src/main/java/org/wordpress/android/ui/people/PersonDetailFragment.java

```diff
@@ -106,4 +159,116 @@ public void setPersonDetails(long personID, int localTableBlogID) {
         mLocalTableBlogID = localTableBlogID;
         refreshPersonDetails();
     }
+
+    // Checks current user's capabilities to decide whether she can change the role or not
+    private void setupRoleContainerForCapability() {
+        Blog blog = WordPress.getBlog(mLocalTableBlogID);
+        Account account = AccountHelper.getDefaultAccount();
+        boolean isCurrentUser = account.getUserId() == mPersonID;
+        boolean canChangeRole = (blog != null) && !isCurrentUser && blog.hasCapability(Capability.PROMOTE_USERS);
+        if (canChangeRole) {
+            mRoleContainer.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    showRoleChangeDialog();
+                }
+            });
+        } else {
+            clearRoleContainerBackground();
+            // Change transparency to give a visual cue to the user that it's disabled
+            mRoleContainer.setAlpha(0.5f);
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    // Remove the selectableItemBackground if the user can't be edited
```

Comment:
May I suggest reinstating this comment back to the line the function is called (line 177 at this point)? The function name is pretty clear and perfectly corresponding to its body, whereas the "if the user can't be edited" part is not implemented here but rather, at the caller's position. Sorry for the mind-blowing nitpicking! 😃 ❤️